File: 643. Maximum Average Subarray I/solution.py
class Solution(object):
  def findMaxAverage(self, nums, k):
    """
    :type nums: List[int]
    :type k: int
    :rtype: float
    """
    maxSum = float(0)
    for num in nums[:k]:
      maxSum += num
      
    totalSum = maxSum
    for index, num in enumerate(nums[k:]):
      totalSum += num - nums[k - k + index] 
      if totalSum > maxSum:
        maxSum = totalSum
        
    average = maxSum / k
    return average

File: 643. Maximum Average Subarray I/solution.java
class Solution {
  public double findMaxAverage(int[] nums, int k) {
    int maxSum = 0;
    for (int index = 0; index < k; index++){
      maxSum += nums[index];
    }

    int totalSum = maxSum;
    for (int index = k; index < nums.length; index++){
      totalSum += nums[index] - nums[index - k];
      if (totalSum > maxSum){
        maxSum = totalSum;
      }
    }

    return (double)maxSum / k;
  }
}